controller/oidc_auth/parse.go

```diff
@@ -0,0 +1,233 @@
+package oidc_auth
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/openziti/edge-api/rest_model"
+	"github.com/openziti/foundation/v2/errorz"
+	"io"
+	"net/http"
+	"reflect"
+	"strings"
+	"unicode/utf8"
+)
+
+type Totp struct {
+	AuthRequestBody
+	Code string `json:"code"`
+}
+
+type updbCreds struct {
+	rest_model.Authenticate
+	AuthRequestBody
+}
+
+func (u *updbCreds) Translate(in string, paths ...string) (string, bool) {
+	if len(paths) > 0 {
+		last := paths[len(paths)-1:][0]
+
+		switch last {
+		case "EnvInfo":
+			return "env" + upperCaseInitial(in), true
+		case "SdkInfo":
+			return "sdk" + upperCaseInitial(in), true
+		}
+	}
+	return "", false
+}
+
+type AuthRequestBody struct {
+	AuthRequestId string `json:"id"`
+}
+
+func (a AuthRequestBody) SetAuthRequestId(id string) {
+	a.AuthRequestId = id
+}
+
+func (a AuthRequestBody) GetAuthRequestId() string {
+	return a.AuthRequestId
+}
+
+var _ AuthRequestIdHolder = (*AuthRequestBody)(nil)
+
+type AuthRequestIdHolder interface {
+	SetAuthRequestId(string)
+	GetAuthRequestId() string
+}
+
+type FieldTranslator interface {
+	Translate(string, ...string) (string, bool)
+}
+
+func MapToStruct(m map[string][]string, dst interface{}) error {
+	translator, _ := dst.(FieldTranslator)
+	return mapToStruct(0, m, dst, translator)
+}
+
```

Comment:
There's a mapstructure library that we already include that does this. I'm not sure if it covers this use case exactly, but might be something to check out.